synth-converter/src/models.rs

```diff
@@ -94,18 +97,22 @@ pub struct Batch {
 }
 
 impl ToGraph for Batch {
-    fn to_triples(&self, subject: SimpleTerm) -> Vec<[SimpleTerm; 3]> {
+    fn to_triples<'a, 'b, 'c>(&'c self, subject: SimpleTerm<'a>) -> Vec<[SimpleTerm<'b>; 3]>
+    where
+        'c: 'b,
+        'a: 'b,
+    {
         let mut triples: Vec<[SimpleTerm; 3]> =
-            [(&rdf::type_, &cat::Batch), (&schema::name, &self::batch_id)]
+            [(&rdf::type_, cat::Batch.as_simple()), (&schema::name, self.batch_id.as_simple())]
```

Comment:
suggestion: This monadic code looks to me very hard to understand.
I dont understand here, since you construct one triple why you go over an iterator, a map and then a collect.
Wouldnt it be easier to read just doing this inplace twice.